packages/ra-core/src/controller/useListController.ts

```diff
@@ -180,11 +180,12 @@ const useListController = <RecordType = Record>(
         get(state.admin.resources, [resource, 'list', 'total'], 0)
     );
 
+    const finalTotal = total || defaultTotal;
```

Comment:
total can be 0, use typeof total === 'undefined' instead

Reply:
You're right!